typing/typecore.ml

```diff
@@ -4729,7 +4766,7 @@ and type_expect_
             { exp with exp_type = instance ty }
         | Tvar _ ->
             let exp = type_exp env sbody in
-            let exp = {exp with exp_type = newty (Tpoly (exp.exp_type, []))} in
+            let exp = {exp with exp_type = newmono exp.exp_type} in
```

Comment:
This feels like a coincidental change since this type is not a function parameter type.